src/coord/src/timestamp.rs

```diff
@@ -431,14 +493,14 @@ impl Timestamper {
     /// Run the update function in a loop at the specified frequency. Acquires timestamps using
     /// either the Kafka topic ground truth
     /// TODO(ncrooks): move to thread local BYO implementation
-    pub fn update(&mut self) {
+    pub fn update(&mut self) -> anyhow::Result<()> {
```

Comment:
_all_ BYO sources - not just ones whose timestamp topic was screwed up - will be completely unusable if this exits (or rather, they will stop updating, and therefore always be stuck at the state before the bad metadata message). I'm not convinced that that is better than panicking. The best thing to do would be to emit an error from the impacted source, so that folks would see it when querying, but other sources would continue to work. But I'm not sure how easy that would be to pipe through everywhere.

Reply:
I think there are multiple issues that this PR raises but does not address:

1. BYO is fundamentally broken, as we flat out ignore errors in reading messages from consistency topics.
2. All BYO sources rely on the same thread for consistency information, even if their sources are not part of the some consistency domain.
3. Errors in reading the consistency topic are not propagated to the sources that rely on them.

When I spoke to @ruchirK, his thinking was that we should exit the timestamper thread on any of these errors. Given that we want to replace the timestamper thread, his didn't think it was worth the effort of plumbing the error messages through to sources.

What do you think?

Reply:
I think it would be strictly better to only stop the topic with the broken messages, rather than stopping the timestamper for all sources.